ShimmerAndroidInstrumentDriver/ShimmerAndroidInstrumentDriver/src/androidTest/java/com/shimmerresearch/android/AAPI_00001_Shimmer_GeneralBluetoothShimmer2R.java

```diff
@@ -0,0 +1,160 @@
+            assertTrue(true);
+        } else {
+            assertTrue(false);
+        }
+
+    }
+
+    @Test   //get firmware version of Shimmer 2R
+    public void test003_GetFW() throws Exception {
+
+        if (shimmer.getFirmwareIdentifier() == ShimmerVerDetails.FW_ID.BTSTREAM) {
+            assertTrue(true);
+        } else {
+            assertTrue(false);
+        }
+        System.out.println("GeneralBluetoothShimmer2RTest C : Firmware version");
+    }
+
+    @Test   //write and read sampling rate of the hardware
+    public void test004_WriteReadSamplingRate() throws Exception {
+
+        double samplingrate;
+        shimmer.readSamplingRate();
+        Thread.sleep(DELAY_DURATION_MS);
+        System.out.println("Current Shimmer 2R sampling rate is:" + shimmer.getSamplingRateShimmer());
+
+        if (shimmer.getSamplingRateShimmer() == 128) {
+            samplingrate = 51.2;
+        } else {
+            samplingrate = 128;
+        }
+        shimmer.writeShimmerAndSensorsSamplingRate(samplingrate);
+        Thread.sleep(DELAY_DURATION_MS);
+        shimmer.readSamplingRate();
+        Thread.sleep(DELAY_DURATION_MS);
+
+
+        if (shimmer.getSamplingRateShimmer() == samplingrate) {
+            assertTrue(true);
+        } else {
+            assertTrue(false);
+        }
+        System.out.println("GeneralBluetoothShimmer2RTest D : Sampling Rate");
+    }
+
+    @Test   //test Shimmer 2R start streaming
+    public void test005_StartStreaming() throws Exception {
+
+        shimmer.startStreaming();
+        Thread.sleep(DELAY_DURATION_MS);
+        if (shimmer.isStreaming()) {
+            assertTrue(true);
+        } else {
+            assertTrue(false);
+        }
+        System.out.println("GeneralBluetoothShimmer2RTest E : Start Streaming");
+    }
+
+    @Test   //test Shimmer 2R stop streaming
+    public void test006_StopStreaming() throws Exception {
+        shimmer.stopStreaming();
+        Thread.sleep(DELAY_DURATION_MS);
+        if (shimmer.isStreaming()) {
+            assertTrue(false);
+        } else {
+            assertTrue(true);
+        }
+        System.out.println("GeneralBluetoothShimmer2RTest F : Stop Streaming");
+    }
+
+    @Test      //test for multiple start and stop streaming
+    public void test007_MultipleStartStopStreaming() throws Exception {
+        for (int i = 1; i < 8; i++) {
+
+            //odd value of i will start streaming
+            if (i % 2 != 0) {
+                shimmer.startStreaming();
+                Thread.sleep(DELAY_DURATION_MS);
+                if (shimmer.isStreaming()) {
+                    assertTrue(true);
+                } else {
+                    assertTrue(false);
+                }
+            }
+            //even value of i will stop streaming
+            else {
+                shimmer.stopStreaming();
+                Thread.sleep(DELAY_DURATION_MS);
+                if (shimmer.isStreaming()) {
```

Comment:
The entire block can be simplified to `assertFalse(shimmer.isStreaming())`

Reply:
good suggestions thanks